pub_key.proto

```diff
@@ -23,38 +23,62 @@ message PubKeyMethod {
     }
 }
 
-message NewPubKeyPayload {
-    enum EntityType {
-        PERSONAL = 0;
-        SERVER = 1;
-    }
-
+message PubKeyStorage {
     enum PubKeyType {
         RSA = 0;
+        ECDSA = 1;
+        EDDSA = 2;
+    }
+
+    enum HashingAlgorithm {
+        SHA256 = 0;
+        SHA512 = 1;
+    }
+
+    message RSAConfiguration {
+        enum Padding {
+            PSS = 0;
+            PKCS = 1;
+        }
+
+        uint64 modulus = 1;
+        bytes exponent = 2;
+        Padding padding = 3;
+    }
+
+    message ECDSAConfiguration {
+        enum EC {
+            SECP256K1 = 0;
+        }
+
+        bytes key = 1;
+        EC ec = 2;
+    }
+
+    message Ed25519Configuration {
+        bytes key = 1;
+    }
+
+    PubKeyType public_key_type = 1;
+    HashingAlgorithm hashing_algorithm = 2;
+
+    bytes entity_hash = 3;
+    bytes entity_hash_signature = 4;
+
+    string owner = 5;
+    bool revoked = 6;
+
+    uint32 valid_from = 7;
+    uint32 valid_to = 8;
+    
+    oneof configuration {
+        RSAConfiguration rsa_configuration = 1;
+        ECDSAConfiguration ecdsa_configuration = 2;
+        EdDSAConfiguration eddsa_configuration = 3;
```

Comment:
"EdDSAConfiguration is undefined". You will get this error if you'll write on JS